app/models/application_form.rb

```diff
@@ -196,6 +210,12 @@ def submitted?
     submitted_at.present?
   end
 
+  def defined_immigration_status?
```

Comment:
I was trying to keep `spec/presenters/register_api/single_application_presenter_spec.rb` from failing, that version causes it to fail 😞

Reply:
It's because it calls `immigration_status`. But your version technically does the same. You have to add `immigration_status` to `PUBLISHED_FIELDS` but I'm not entirely sure what the implications of that are. CC @JR-G @tomas-stefano

Reply:
Going into published fields means that will update all application choices if there is a update in the field, therefore they will bump in the order those choices shows into the API. But the moment you add to published fields the specs will verify if the presenter is query for that field and fails otherwise. Is that the desired output? @JR-G if so I'd say @vassyz suggestion is good.
If the above statement is true you can also do `immigration_status.in?(ApplicationForm.immigration_statuses.keys - %w[other])`

Reply:
That's right – but I think in this scenario that's an acceptable change, it would make sense to have it bumped if there immigration status has changed